src/main/java/com/uid2/admin/cloudEncryption/ExpiredKeyCountRetentionStrategy.java

```diff
@@ -0,0 +1,58 @@
+package com.uid2.admin.cloudEncryption;
+
+import com.uid2.admin.store.Clock;
+import com.uid2.shared.model.CloudEncryptionKey;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+// Keep up to `expiredKeysToRetain` most recent expired keys
+// Considers all keys passed, doesn't do grouping by site (handled upstream)
+public class ExpiredKeyCountRetentionStrategy implements CloudKeyRetentionStrategy {
+    private final Clock clock;
+    private final int expiredKeysToRetain;
+
+    public ExpiredKeyCountRetentionStrategy(Clock clock, int expiredKeysToRetain) {
+        this.clock = clock;
+        this.expiredKeysToRetain = expiredKeysToRetain;
+    }
+
+    @Override
+    public Set<CloudEncryptionKey> selectKeysToRetain(Set<CloudEncryptionKey> keysForSite) {
```

Comment:
This could be a lot simpler if we just sort by Created descending and take the top expiredKeysToRetain? 

I thought we didn't have a concept of "expired" keys?

Reply:
Perhaps expiredKeysToRetain should just be called keysToRetain

Reply:
That is if we completely discount possibility of creating keys for the future. I guess we could do that with the new design. There is still code that creates some future keys on creation/update of operators in `CloudEncryptionKeyManager` - I'll remove it there as part of the next MR. Probably not a serious problem though - happy to change this part.

Reply:
As a compromise and safeguard, I'll make this discard any keys which are in the future.